src/gen_apple/ga_dev_xbee.cpp

```diff
@@ -1,26 +1,117 @@
 #include "ga_dev_xbee.h"
 
+struct XBee xbee;
+
 void ga_dev_xbee_open(void)
 {
     soft_serial.begin(9600);
-    xbee.begin(soft_serial);
+    xbee.serial = &soft_serial;
+    xbee.api_id = ZB_TX_REQUEST;
+    xbee.frame_id = DEFAULT_FRAME_ID;
+    xbee.broadcast_range = ZB_BROADCAST_RADIUS_MAX_HOPS;
+    xbee.option = ZB_TX_UNICAST;
+    xbee.addr16 = ZB_BROADCAST_PACKET;
+    xbee.msb = 0;
+    xbee.lsb = 0;
 }
 
-int ga_dev_xbee_avail(void)
-{
+int ga_dev_xbee_avail(void) {
     return Serial.available();
 }
 
-int ga_dev_xbee_read(void)
-{
+int ga_dev_xbee_read(void) {
     return Serial.read();
 }
 
-void ga_dev_xbee_write(uint8_t *data, int data_len)
-{
-    XBeeAddress64 addr64 = XBeeAddress64(0, 0);
-    ZBTxRequest zbtx = ZBTxRequest(addr64, data, data_len);
+void ga_dev_xbee_write(uint8_t val) {
+    xbee.serial -> write(val);
+}
+
+uint8_t ga_dev_xbee_frame_length(int data_len) {
+    return ZB_TX_API_LENGTH + data_len;
+}
+
+uint8_t ga_dev_xbee_frame_data(uint8_t* data, int pos) {
```

Comment:
This function also uses a bunch of tabs. 

Did you follow this style from a library? Maybe a switch statement would look better, but it might not matter too much.